tests/test.py

```diff
@@ -102,6 +102,48 @@ def test_i2c_set_bus_voltage_or_init_bus_are_equivalent(self):
         except Exception as e:
             self.fail(f"I2C read raised an exception {e}")
 
+    def test_i2c_set_pull_up_resistor(self):
+        if self.use_simulator:
+            self.skipTest("For real device only")
+
+        i2c = self.device.create_interface("i2c")
+
+        (success, message) = i2c.set_pull_up_resistors(150)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 150 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(220)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 220 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(330)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 330 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(470)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 470 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(680)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 680 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(1000)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 1000 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(1500)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 1500 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(2200)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 2200 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(3300)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 3300 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(4700)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 4700 Ohm failed: {message}")
+        (success, message) = i2c.set_pull_up_resistors(10000)
+        self.assertTupleEqual((True, None), (success, message), f"Setiing 10000 Ohm failed: {message}")
+
+    def test_i2c_set_pull_up_resistor_exception_unsupported_value(self):
```

Comment:
Just a comment (that gpt taught me). If you want to run some specific tests instead of all the unittest, you can do the following, where I run the 2 new tests you want to add:
```
python -m unittest -k "test_i2c_set_pull_up_resistor or test_i2c_set_pull_up_resistor_exception_unsupported_value"
```